tests/test_cli_compile.py

```diff
@@ -946,6 +946,54 @@ def test_generate_hashes_with_long_annotations(runner):
     )
 
 
+@pytest.mark.network
+def test_generate_hashes_with_line_style_annotations(runner):
+    with open("requirements.in", "w") as fp:
+        fp.write("Django==1.11.29\n")
+        fp.write("django-debug-toolbar==1.11\n")
+        fp.write("django-storages==1.9.1\n")
+        fp.write("django-taggit==0.24.0\n")
+        fp.write("pytz==2020.4\n")
+        fp.write("sqlparse==0.3.1\n")
+
+    out = runner.invoke(cli, ["--generate-hashes", "--annotation-style", "line"])
+    assert out.stderr == dedent(
+        f"""\
+        #
+        # This file is autogenerated by pip-compile with python \
+{sys.version_info.major}.{sys.version_info.minor}
+        # To update, run:
+        #
+        #    pip-compile --annotation-style=line --generate-hashes
+        #
+        django==1.11.29 \\
+            --hash=sha256:014e3392058d94f40569206a24523ce254d55ad2f9f46c6550b0fe2e4f94cf3f \\
+            --hash=sha256:4200aefb6678019a0acf0005cd14cfce3a5e6b9b90d06145fcdd2e474ad4329c
```

Comment:
There used to be a backslash at the end of each hash line

Reply:
Yeah, but was that a mistake, or unnecessary? As in, the next line really has no business or need to be on a line continuation?

Just let me know what makes sense.

Reply:
And it _must_ not have a continuation if there's no annotation.

Reply:
Also, thanks for taking a look! Looking forward to this feature's return.

Reply:
In the spirit of line-oriented parsing, if someone uses a simple `sed` expression to strip comment lines, we don't need the complication of breaking the resulting syntax because one req continues into another.

Reply:
The backlash was technically unnecessary, as it preceded a comment line, but you could argue that it was making one logical line with all hashes and the `via` comment, and more importantly it avoided unnecessary diff noise when new hashes were added.  If an option is added to restore the older, nicer format, then please restore it fully :)

Reply:
For me personally the point here is to be friendly to line oriented processing, so I don't wish to make that harder with the pre-comment continuation. But I will make it so if there's a consensus.

Reply:
I don’t fully understand your use cases.  Either line-oriented tools work with physical lines, in which case my request doesn’t change anything, or they interpret backslashes to make logical lines and we’ll still have one line per dependency.

If you want to remove hashes, it would be simpler to call pip-compiles with different options.

My request is to keep the same format that we had before to avoid huge diffs.  This includes the backslash on each hash line 🙂

Reply:
I'm on mobile and busy right now, but the example is removing comments, not hashes. A naive comment remover would combine non comment lines, resulting in invalid syntax.

Reply:
So an explicit example of this breaking naive line-oriented expectations:

`reqs.txt`:
```ini
django-debug-toolbar==1.11 \
    --hash=sha256:89d75b60c65db363fb24688d977e5fbf0e73386c67acf562d278402a10fc3736 \
    --hash=sha256:c2b0134119a624f4ac9398b44f8e28a01c7686ac350a12a74793f3dd57a9eea0 \
    # via -r requirements.in
django-storages==1.9.1 \
    --hash=sha256:3103991c2ee8cef8a2ff096709973ffe7106183d211a79f22cf855f33533d924 \
    --hash=sha256:a59e9923cbce7068792f75344ed7727021ee4ac20f227cf17297d0d03d141e91 \
    # via -r requirements.in
```

```console
$ grep -Ev '^(\s*#|$)' reqs.txt
```

```ini
django-debug-toolbar==1.11 \
    --hash=sha256:89d75b60c65db363fb24688d977e5fbf0e73386c67acf562d278402a10fc3736 \
    --hash=sha256:c2b0134119a624f4ac9398b44f8e28a01c7686ac350a12a74793f3dd57a9eea0 \
django-storages==1.9.1 \
    --hash=sha256:3103991c2ee8cef8a2ff096709973ffe7106183d211a79f22cf855f33533d924 \
    --hash=sha256:a59e9923cbce7068792f75344ed7727021ee4ac20f227cf17297d0d03d141e91 \
```

Reply:
Thanks for explaining the use case.  I don’t see any need for me to use such line procesing, and as I explained see more value in lines being diff-friendly and keeping the format that I currently have in all repositories (having pinned pip-tools for months now).

Reply:
To be clear, this concern is only about a one-time commit per `txt`, right? As in, the diffs won't be any bigger on an ongoing basis, only potentially bigger one time while getting over the `pip-tools` update hump?

Reply:
I'll explicitly note that adding that trailing slash would also mean unnecessarily larger diffs each time annotations are turned on or off. And this is not limited to a one-time `pip-tools` update hump.

Reply:
I agree with @merwok.
This PR is about adding a new (well old) output format. 
Please don't mix in a change to the current format.

Reply:
Yes, the concern is about one commit.  I use pip-tools in many projects and don’t have a need to change options often.  I always want annotations, and I typically have hashes on from the start or enable the option once then keep it.  But! you’re the one doing the work here to restore a style close to what we had, so if I’m not convincing you then I’ll say thanks and take the one-time useless commit in my repos 🙂

Reply:
Thanks. I want to emphasize that while I am still against it, I'm not rejecting this yet, especially since @ulope is in agreement with you. 

Thank you for laying out the key details with me.

I just want to make sure the potential advantages and disadvantages are written out for a proper consensus to be reached.

Reply:
I've added the line continuations; please check if they're working as expected, thanks!